frontend/containers/image-gallery/component.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, useState } from 'react';
+
+import { useIntl } from 'react-intl';
+
+import cx from 'classnames';
+
+import Image from 'next/image';
+
+import Button from 'components/button';
+import Carousel, { Slide } from 'components/carousel';
+import Modal from 'components/modal';
+
+import { ImageGalleryProps } from '.';
+
+export const ImageGallery: FC<ImageGalleryProps> = ({ images }) => {
+  // Sets the index of the image displayed on carousel. The default is undefined for not displaing any image.
+  const [active, setActive] = useState<number>();
+  const { formatMessage } = useIntl();
+
+  return (
+    <div area-hidden>
+      <ul className="flex gap-1">
+        {images.map(({ file: { small } }, index) => {
+          const selected = active === index;
+          return (
+            <li key={small} className="inline">
+              <Button
+                theme={selected ? 'secondary-green' : 'naked'}
+                className={cx('block w-8 h-8 px-0 py-0 rounded-none  transition-all', {
+                  'brightness-125': selected,
+                  'brightness-75': !selected,
+                })}
+                onClick={() => setActive(index)}
+                title={formatMessage({ defaultMessage: 'Open image', id: 'TLn/wq' })}
+              >
+                <Image
+                  src={small}
+                  alt={formatMessage({ defaultMessage: 'Project image', id: 's07dAT' })}
+                  width={32}
+                  height={32}
+                  objectFit="cover"
+                />
+              </Button>
+            </li>
+          );
+        })}
+      </ul>
+      <Modal
+        onDismiss={() => setActive(undefined)}
+        title={formatMessage({ defaultMessage: 'Image gallery', id: 'K+Q04b' })}
+        open={typeof active === 'number'}
+        dismissable={true}
+        size="wide"
+        scrollable={false}
+      >
+        <Carousel defaultSlide={active} className="h-full sm:mx-12">
+          {images.map(({ file: { original, medium } }) => (
+            <Slide key={original} className="flex items-center justify-center w-full h-full">
+              <div className="w-full">
+                <Image
+                  src={medium}
```

Comment:
When transitioning images, they take quite a while to load. 

https://user-images.githubusercontent.com/6273795/170469088-25a6af48-77d4-45b9-9310-b13140b16ace.mov

It may be worth it to preload them with [`priority={true}`](https://nextjs.org/docs/api-reference/next/image#priority).

Reply:
I tested dragging the slides on chrome dev tools simulation of a touch screen and the slide works fine. It just doesn't work well when clicking. 

I added the priority!